src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -185,6 +211,18 @@ document::value list_databases(entity::map& map,
     return result.extract();
 }
 
+document::value list_database_names(entity::map& map, const std::string& object) {
+    auto cursor = map.get_client(object).list_database_names();
```

Comment:
It is not entirely clear if/when either the `session` or `filter` arguments should be forwarded to the relevant operations. As currently written, `list_collection_names` above conditionally forwards both `session` and `filter` as done in `list_collections`, but unlike `list_databases`, `list_database_names` does not currently have an overload that accepts a `client_session`. Should such an overload be added to `mongocxx::client` to support passing a `session` argument (in addition to a `filter` argument)?

Also note, `list_databases` currently doesn't have an overload that accepts _both_ a session and an opts argument either.

Reply:
> Should such an overload be added to `mongocxx::client` to support passing a `session` argument (in addition to a `filter` argument)?

I think "yes". I expect not having an overload accepting `session` was an oversight.

`session` is not mentioned in the pspecification describing listDatabases or listDatabaseNames](https://github.com/mongodb/specifications/blob/master/source/enumerate-databases.rst#enumerating-database-names). But I see no reason to exclude `session`. All drivers except Rust [appear to accept session in listDatabaseNames](https://docs.google.com/spreadsheets/d/1JEQfCE1CnAt47tqFkDgrJ7eNenM_0bMbvgNYK-y8iME/edit#gid=0).

> Also note, `list_databases` currently doesn't have an overload that accepts _both_ a `session` and `opts` argument either.

I expect that was also an oversight.

I suggest adding these overloads as part of this PR, or filing a CXX ticket to request the additions.